### PR TITLE
add job token auth and job mapping

### PR DESCRIPTION
**Pull request by aaannz**
This adds simple job token authentication. Token is stored as JOBTOKEN key in JobSettings table for running jobs only (added during job_grab and removed when job_done and cloning).

For testing purposes added route /api/v1/whoami returning jobid.

added function job_set_settings($jobid, $key, [$value]) for adding (and removing if $value is undef) job settings data

### CODE REVIEW

**Comment by coolo** [comment]
I guess I should accelerate the jobs as DBIx work - this is yet another job_* function in scheduler, when I think it should be a set_property in the Job result. But this is majorly inconsistent atm ;(

But I don't object, just saying - looks good to me other than that.

**Comment by ancorgs** [comment]
In my opinion, Scheduler and Utils should tend to shrink and not to grow with every commit. I would feel better if the code you added to Scheduler would leave in the Jobs and/or JobSettings models (i.e. under OpenQA::Schema).

**Comment by aaannz** [comment]
Completely agree, Scheduler.pm refactoring is IMO long overdue. I will redo job settings into Job result class.

File: lib/OpenQA/Scheduler.pm [hunk 1]
```diff
@@ -634,23 +634,54 @@ sub job_grab {
...
+            state => OpenQA::Schema::Result::Jobs::RUNNING,
+            worker_id => $workerid,
+        }
+    )->single;
+    return {} unless ($job);
+
+    # generate jobtoken for test access
+    my @chars = ("A".."Z", "a".."z", '0'..'9');
+    my $token;
+    $token .= $chars[rand @chars] for 1..32;
```
Comment (lnussel) [inline lib/OpenQA/Scheduler.pm]: there's a rndstr function already in lib/db_helpers.pm resp lib/OpenQA/Plugin/Helpers.pm

**Comment by lnussel** [comment]
doesn't this make JOBTOKEN public for anyone querying job information? That seems to defeat the purpose then. What about storing the current token in WorkerProperties instead?

**Comment by coolo** [comment]
we can blank specific values in the job view - this might be necessary for other things in the future too

**Comment by aaannz** [comment]
Do we pass worker properties to vars.json? Since I need this value to get to actual running test. But I still like it to stick with specific job.

OTOH I am not a big fan of custom field blanking, it smells like to much hard coded stuff on many places.

**Comment by coolo** [comment]
then add a column to job_settings to mark a setting as private

**Comment by coolo** [comment]
please make an issue for that privacy leak and we care about that later

**Comment by aaannz** [comment]
Done, action 6344

**Comment by lnussel** [comment]
Ondřej Holeček schrieb:
> Do we pass worker properties to vars.json? Since I need this value to
> get to actual running test. But I still like it to stick with specific
> job.

The setting can be synthesized into the json only when the worker grabs
the job so only the worker actually sees it.

cu
Ludwig

-- 
  (o_   Ludwig Nussel
  //\
  V_/_  http://www.suse.de/
SUSE Linux GmbH, GF: Felix Imendörffer, Jane Smithard, Jennifer
Guild, Dilip Upmanyu, Graham Norton, HRB 21284 (AG Nürnberg)
Maxfeldstraße 5; 90409 Nürnberg; Germany

**Pull request merged**